hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/UpsertPartitioner.java

```diff
@@ -86,16 +87,21 @@ public class UpsertPartitioner<T> extends SparkHoodiePartitioner<T> {
   private HashMap<Integer, BucketInfo> bucketInfoMap;
 
   protected final HoodieWriteConfig config;
+  private final WriteOperationType operationType;
 
   public UpsertPartitioner(WorkloadProfile profile, HoodieEngineContext context, HoodieTable table,
-      HoodieWriteConfig config) {
+                           HoodieWriteConfig config, WriteOperationType operationType) {
     super(profile, table);
     updateLocationToBucket = new HashMap<>();
     partitionPathToInsertBucketInfos = new HashMap<>();
     bucketInfoMap = new HashMap<>();
     this.config = config;
+    this.operationType = operationType;
     assignUpdates(profile);
-    assignInserts(profile, context);
+    long totalInserts = profile.getInputPartitionPathStatMap().values().stream().mapToLong(stat -> stat.getNumInserts()).sum();
+    if (!WriteOperationType.isPreppedWriteOperation(operationType) || totalInserts > 0) { // skip if its prepped write operation. or if totalInserts = 0.
+      assignInserts(profile, context);
```

Comment:
So you are assuming the prepped operation is always upserts.

Reply:
this is probably because the record size estimation is done only when assigning inserts.

Reply:
yeah, just left some comments to make it more clear.

Reply:
yes. thats the main purpose of prepped write operation. we expect the location to be set already in them and so we do not invoke tag locations. and hence assignInserts() is unnecessary overhead and is a no-op.

Reply:
@nsivabalan should the if condition be a &&? Its still calling assignInserts with MDT.